Automattic/woocommerce-payments
# Prevent possible fatal when using `get_edit_post_link` filter

Fixes #7881.

#### Changes proposed in this Pull Request

When using the `get_edit_post_link` it can end up receiving another input rather than a valid URL either from a previous callback up in the chain, or the original call. The [official docs of the filter](https://developer.wordpress.org/reference/functions/get_edit_post_link/) says it can return `null` if the post type does not exist or does not allow an editing UI.

With that said, we don't have a reason to define a strict `string` return in the filter callback. I also added an extra verification to validate the URL and return `null` if it's not valid. The extra verification is mostly an attempt to prevent bad code from messing with the final output.









#### Testing instructions






1. Add the following custom code to simulate a bad third-party code. I used the Code Snippets extension to do that.
```php
add_filter( 'get_edit_post_link', function( $url ) {
} );
```
2. Enable "WP post order storage" in `WooCommerce > Settings > Advanced > Features`.
3. Go to `WooCommerce > Orders`.
4. You should **not** see a fatal error.

Note on step 3: if you don't have any orders placed, going to the store page, adding an item to the cart and going to the checkout page also triggers the relevant code.

-------------------


**Post merge**
